File: api/src/main/java/net/blueshell/api/dto/AddressDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.validation.address.ValidCountryCode;

import java.time.Instant;

@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
@Data
@Schema(name = "Address")
public class AddressDTO extends BaseDTO {
    private Long id;

    @NotEmpty
    @ValidCountryCode
    private String country;

    @NotEmpty
    private String city;

    @NotEmpty
    private String street;

    @NotEmpty
    private String houseNumber;

    @NotEmpty
    private String zipCode;
}

File: api/src/main/kotlin/net/blueshell/api/dto/AddressDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotEmpty
import net.blueshell.api.dto.base.AuditedAutoIdDTO

@Schema(name = "Address")
data class AddressDTO(
    @field:NotEmpty
    @field:net.blueshell.api.validation.address.ValidCountryCode
    var country: String? = null,

    @field:NotEmpty
    var city: String? = null,

    @field:NotEmpty
    var street: String? = null,

    @field:NotEmpty
    var houseNumber: String? = null,

    @field:NotEmpty
    var zipCode: String? = null
) : AuditedAutoIdDTO()
